### PR TITLE
BTRX - 773 - Convert "Project Report" page

### PR DESCRIPTION
**Pull request by LeoForconesi**
## Addresses
[BTRX-773](https://broadinstitute.atlassian.net/browse/BTRX-773) : Convert "Project Report" page

## Changes
- Constants rename for qa report
- Add Project report link to Qa Event Report
- Fix project status box when redirected from Project Report
- Add a new api resource to return an issue event list

## Testing
- From QA Event report, access to a project key link in "Project" table's column. (if this is opened in another browser window, it should still be working)
- In Project Report page, test any table functionallity, and the header project key link.
- When redirected from Project Report header link, the project status box should display its status.
---

- [ ] **Submitter**: Verify all tests go green, including CI tests
- [ ] **Submitter**: Get a thumb from Belatrix
- [ ] **Submitter**: Get a thumb from @rushtong
- [ ] **Submitter**: Merge to develop using github's "Squash and Merge" feature
- [ ] **Submitter**: Delete branch after merge

### CODE REVIEW

File: grails-app/services/org/broadinstitute/orsp/StatusEventService.groovy [hunk 1]
```diff
@@ -27,21 +27,23 @@ class StatusEventService {
...
-                        ' where e.project_key = :projectKey ' +
-                        ' and e.event_type in :typeList '
-        final sqlQuery = session.createSQLQuery(query)
-        final results = sqlQuery.with {
-            addEntity(Event)
-            setString('projectKey', projectKey)
-            setParameterList('typeList', TYPES*.name())
-            list()
+    List<StatusEventDTO> getStatusEventsForProject(final String projectKey) {
+        if(projectKey) {
```
Comment (rushtong) [inline grails-app/services/org/broadinstitute/orsp/StatusEventService.groovy]: nit: spacing
```suggestion
        if (projectKey) {
```

File: src/main/webapp/qaReport/ProjectReport.js [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+  }
+
+  componentDidMount = async () => {
+    this.props.showSpinner();
+    try {
+      const result = await Reports.getQaEventReportForProject(get(qs.parse(this.props.location.search), 'projectKey', ''));
+      this.setState({ data: result.data },
+        ()=> this.props.hideSpinner()
+      );
+    } catch(error) {
```
Comment (rushtong) [inline src/main/webapp/qaReport/ProjectReport.js]: nit: spacing:
```suggestion
    } catch (error) {
```

File: src/main/webapp/util/ReportsConstants.js [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+  dataField: 'projectKey',
+  text: 'Project',
+  sort: true,
+  formatter: (cell, row, rowIndex, colIndex) =>
+    h(Link,{
+      to: '/statusEvent/projectReport?projectKey=' + row.projectKey
+        //{
+        // pathname:'/statusEvent/projectReport?projectKey=' + row.projectKey
+        // state: {projectKey: row.projectKey}
+      //}
```
Comment (rushtong) [inline src/main/webapp/util/ReportsConstants.js]: Intentional? Remove if not needed.

**Review by rushtong — APPROVED** [review]
Looks good. Minor nits inline.

File: grails-app/services/org/broadinstitute/orsp/StatusEventService.groovy [hunk 4]
```diff
@@ -27,21 +27,23 @@ class StatusEventService {
...
+            final session = sessionFactory.currentSession
+            final String query =
+                    ' select * from event e where e.project_key = :projectKey ' +
```
Comment (dags) [inline grails-app/services/org/broadinstitute/orsp/StatusEventService.groovy]: trim spaces

File: src/main/webapp/qaReport/ProjectReport.js [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+    super(props);
+    this.state = {
+      data: []
+    };
+  }
+
+  componentDidMount = async () => {
+    this.props.showSpinner();
+    try {
+      const result = await Reports.getQaEventReportForProject(get(qs.parse(this.props.location.search), 'projectKey', ''));
```
Comment (dags) [inline src/main/webapp/qaReport/ProjectReport.js]: extract this to a local function

File: src/main/webapp/util/ReportsConstants.js (hunk 3 above)
Comment (LeoForconesi) [reply src/main/webapp/util/ReportsConstants.js]: This has been removed.

File: grails-app/services/org/broadinstitute/orsp/StatusEventService.groovy (hunk 4 above)
Comment (LeoForconesi) [reply grails-app/services/org/broadinstitute/orsp/StatusEventService.groovy]: Done.

**Review by dags — APPROVED** [review]

**Pull request merged**